cmake/modules/AddCephTest.cmake

```diff
@@ -3,8 +3,8 @@
 #adds makes target/script into a test, test to check target, sets necessary environment variables
 function(add_ceph_test test_name test_path)
   add_test(NAME ${test_name} COMMAND ${test_path})
-  add_dependencies(check ${test_name})
-  set_property(TEST 
+  add_dependencies(tests ${test_name})
```

Comment:
@tchaikov following up on our irc discussion, since i think we were misunderstanding each other

this line is what was causing 'make check' to build all of the tests before running ctest, so removing the dependencies from 'check' would be a regression ('make clean && make check' would not rebuild the tests). all i'm looking for is a `DEPENDS tests` in the `add_custom_target(check ...)`, so it continues to pull in these dependencies

i understand that other tests will need to add more dependencies for things it needs at runtime, and i'm not suggesting we block this pr waiting for those. i just want to avoid a regression in the 'check' target itself. if 'check' depends on 'tests', then it will pull in any dependencies we add later on